docker/build.sh

```diff
@@ -0,0 +1,81 @@
+# conditions and the following disclaimer in the documentation and/or other
+# materials provided with the distribution.
+# * Neither the name of Caltech nor its operating division, the Jet Propulsion
+# Laboratory, nor the names of its contributors may be used to endorse or
+# promote products derived from this software without specific prior written
+# permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+# ----------------------------------------------------------------------------------------------
+# This script is used to build the docker image for the Registry Loader with a single command.
+#
+# Both Harvest and Registry Manager tools are included in this Registry Loader docker image.
+# This script reads required configurations from a properties file passed as an argument.
+#
+# Usage: ./build.sh docker.properties
+#
+# ----------------------------------------------------------------------------------------------
+
+# Check if a properties file name is provided as an argument to this script
+if [ -z "$1" ]; then
+  echo -e "\nPlease provide a properties file as an argument to $0"
+  echo -e "\tUsage: $0 docker.properties \n"
+  exit 1
+fi
+
+PROPERTY_FILE=$1
+
+# Check if the provided properties file exists
+if [ ! -f "$PROPERTY_FILE" ]; then
+    echo -e "\nThe file $PROPERTY_FILE does not exist."
+    echo -e "\tPlease provide an existing properties file as an argument to $0"
+    echo -e "\tUsage: $0 docker.properties \n"
+    exit 1
+fi
+
+# Returns the value for a given property key
+function getProperty {
+   PROPERTY_KEY=$1
+   PROPERTY_VALUE=$(cat "$PROPERTY_FILE" | grep -w "$PROPERTY_KEY" | cut -d'=' -f2)
+   echo "$PROPERTY_VALUE"
+}
+
+# Read property values from the properties file
+echo "Reading properties from $PROPERTY_FILE"
+HARVEST_VERSION=$(getProperty "harvest.version")
+REG_MANAGER_VERSION=$(getProperty "reg.manager.version")
+DOCKER_IMAGE_NAME_TAG=$(getProperty "docker.image.name.tag")
+
+echo "HARVEST_VERSION  = $HARVEST_VERSION"
+echo "REG_MANAGER_VERSION = $REG_MANAGER_VERSION"
+echo "DOCKER_IMAGE_NAME_TAG = $DOCKER_IMAGE_NAME_TAG"
+
+# Execute Docker build with build arguments
+docker build --build-arg harvest_version="$HARVEST_VERSION" \
```

Comment:
I like to be explicit and say `docker image build` to help underscore the difference between containers and images but that's just my taste 😋

Reply:
Agreed. Used `docker image build` instead of `docker build`.